samples/subsys/usb/midi/README.rst

```diff
@@ -0,0 +1,84 @@
+.. zephyr:code-sample:: midi
+   :name: USB MIDI device class
+   :relevant-api: usbd_api usb_midi input_events
+
+   Implements a simple USB-MIDI loopback and keyboard device.
+
+Overview
+********
+
+This sample demonstrates how to implement a USB MIDI device. It can run on
+any board with a USB device controller. This sample sends all MIDI1 messages
+sent to the device back to the host. In addition, presses and release on
+input keys (such as the board user buttons) are sent as MIDI1 note on and
+note off events.
+
+The application exposes a single USB-MIDI interface with a single bidirectional
+group terminal. This allows exchanging data with the host on a "virtual wire"
+that carries MIDI1 messages, pretty much like a standard USB-MIDI in/out adapter
+would provide. The loopback acts as if a real MIDI cable was connected between
+the output and the input, and the input keys act as a MIDI keyboard.
+
+Building and Running
+********************
+
+The code can be found in :zephyr_file:`samples/subsys/usb/midi`.
+
+To build and flash the application:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/usb/midi
+   :board: nucleo_f429zi
+   :goals: build flash
+   :compact:
+
+Using the MIDI interface
+************************
+
+Once this sample is flashed, connect the device USB port to a host computer
+with MIDI support. For example, on Linux, you can use alsa to access the device:
+
+.. code-block:: console
+
+  $ amidi -l
```

Comment:
It does not work, I guess there is something wrong in the interface descriptors.

Reply:
On which board did you try ? On my side, the sample just works out of the box on these 2 boards :

- nucleo_f429zi (https://docs.zephyrproject.org/latest/boards/st/nucleo_f429zi/doc/index.html)
- stm32f4_disco (https://docs.zephyrproject.org/latest/boards/st/stm32f4_disco/doc/index.html)

Both are based on stm32f4, and only support Full-Speed USB; so I haven't been able to test on High-Speed, this may be your case ?

Reply:
reel board, bu tit is quite simple function, should work with any board in the tree. I will check again later to see what is wrong.

Reply:
When you retry, could you please also share:

- What host OS are you using (`uname -r`) ?
- What version of Alsa are you using (`amidi --version`) ?
- Are there some useful kernel messages during device enumeration (`dmesg`) ?

In my case:

- `6.11.10-2-MANJARO`
- `amidi version 1.2.13`

```
kernel: usb 1-3.4: new full-speed USB device number 121 using xhci_hcd
kernel: usb 1-3.4: New USB device found, idVendor=2fe3, idProduct=0010, bcdDevice= 4.00
kernel: usb 1-3.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
kernel: usb 1-3.4: Product: USBD MIDI Sample
kernel: usb 1-3.4: Manufacturer: Zephyr Project
kernel: usb 1-3.4: SerialNumber: 31383336303851060041003B
```